pylint_odoo/test_repo/broken_module/models/broken_model.py

```diff
@@ -78,3 +78,33 @@ def my_method9(self):
     def my_method10(self):
         # A example of built-in raise without parameters
         raise ZeroDivisionError
+
+    def sql_method(self, ids, cr):
+        # This is the better way and should not be detected
+        self._cr.execute(
+            'SELECT name FROM account WHERE id IN %s', (tuple(ids),))
+        self.env.cr.execute(
+            'SELECT name FROM account WHERE id IN %s', (tuple(ids),))
+        cr.execute(
+            'SELECT name FROM account WHERE id IN %s', (tuple(ids),))
+
+    def sql_injection_method(self, cr, ids):
+        # SQL injection, bad way
+        self._cr.execute(
+            'SELECT name FROM account WHERE id IN %s' % (tuple(ids),))
+        self.env.cr.execute(
+            'SELECT name FROM account WHERE id IN %s' % (tuple(ids),))
+        cr.execute(
+            'SELECT name FROM account WHERE id IN %s' % (tuple(ids),))
+
+    def sql_injection_method3(self, ids, cr2):
+        # This cr.execute2 or cr2.execute should not be detected
+        self._cr.execute2(
+            'SELECT name FROM account WHERE id IN %s' % (tuple(ids),))
+        cr2.execute(
+            'SELECT name FROM account WHERE id IN %s' % (tuple(ids),))
+
+    def sql_injection_method5(self, ids):
+        var = 'SELECT name FROM account WHERE id IN %s'
+        values = ([1, 2, 3, ], )
+        self._cr.execute(var % values)  # sql injection too
```

Comment:
@moylop260 how does this work exactly. I guess these methods will be called somehow in unit tests ? For now, I don't get how you can be sure that sql_method will not be detected and sql_injection_method will be.